staticd/static_srv6.c

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * STATICd - Segment Routing over IPv6 (SRv6) code
+ */
+#include <zebra.h>
+
+#include "vrf.h"
+#include "nexthop.h"
+
+#include "static_routes.h"
+#include "static_srv6.h"
+#include "static_vrf.h"
+#include "static_zebra.h"
+
+/*
+ * List of SRv6 SIDs.
+ */
+struct list *srv6_locators = NULL;
+struct list *srv6_sids = NULL;
+
+DEFINE_MTYPE_STATIC(STATIC, STATIC_SRV6_LOCATOR, "Static SRv6 locator");
+DEFINE_MTYPE_STATIC(STATIC, STATIC_SRV6_SID, "Static SRv6 SID");
+
+/*
+ * Convert SRv6 behavior to human-friendly string.
+ */
+const char *static_srv6_sid_behavior2str(enum static_srv6_sid_behavior_t behavior)
+{
+	switch (behavior) {
+	case STATIC_SRV6_SID_BEHAVIOR_END:
+		return "End";
+	case STATIC_SRV6_SID_BEHAVIOR_END_X:
+		return "End.X";
+	case STATIC_SRV6_SID_BEHAVIOR_END_DT6:
+		return "End.DT6";
+	case STATIC_SRV6_SID_BEHAVIOR_END_DT4:
+		return "End.DT4";
+	case STATIC_SRV6_SID_BEHAVIOR_END_DT46:
+		return "End.DT46";
+	case STATIC_SRV6_SID_BEHAVIOR_UN:
+		return "uN";
+	case STATIC_SRV6_SID_BEHAVIOR_UA:
+		return "uA";
+	case STATIC_SRV6_SID_BEHAVIOR_UDT6:
+		return "uDT6";
+	case STATIC_SRV6_SID_BEHAVIOR_UDT4:
+		return "uDT4";
+	case STATIC_SRV6_SID_BEHAVIOR_UDT46:
+		return "uDT46";
+	case STATIC_SRV6_SID_BEHAVIOR_UNSPEC:
+		return "unspec";
+	}
+
+	return "unspec";
+}
+
+/*
+ * When an interface is enabled in the kernel, go through all the static SRv6 SIDs in
+ * the system that use this interface and install/remove them in the zebra RIB.
+ *
+ * ifp   - The interface being enabled
+ * is_up - Whether the interface is up or down
+ */
+void static_ifp_srv6_sids_update(struct interface *ifp, bool is_up)
+{
+	struct static_srv6_locator *locator;
+	struct static_srv6_sid *sid;
+	struct listnode *node;
+
+	if (!srv6_sids || !ifp)
+		return;
+
+	zlog_info("Interface %s %s. %s SIDs that depend on the interface",
+		  (is_up) ? "enabled" : "disabled", (is_up) ? "Removing" : "disabled", ifp->name);
+
+	/* iterate over the list of SRv6 SIDs and remove the SIDs that use this
+    * VRF from the zebra RIB */
```

Comment:
format must be manually done on comments as 'git clang-format' does not know how to apply it : tab spaces before `\` and `*`

Reply:
thanks